pkgs/development/libraries/gobject-introspection/default.nix

```diff
@@ -27,6 +27,12 @@
 # it may be worth thinking about using multiple derivation outputs
 # In that case its about 6MB which could be separated
 
+let
+  python = python3.withPackages (pp: [
```

Comment:
If this is a build time dep then use python3.pythonForBuild

Reply:
It is a runtime dependency that is checked at build time:

https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/6a934c41190740b4e69bdeaae2e7b57bb5e33ac8/meson.build#L201-202